idea/data-storage/src/gear/gear-event-listener.ts

```diff
@@ -276,30 +257,45 @@ export class GearEventListener {
             data: { id, change },
           } = event.event as CodeChanged;
           const codeStatus = change.isActive ? CodeStatus.ACTIVE : change.isInactive ? CodeStatus.INACTIVE : null;
-
-          codes.push({
+          const updateCodeInput = {
             id: id.toHex(),
             genesis: this.genesis,
             status: codeStatus,
             timestamp,
             blockHash: block.createdAtHash.toHex(),
             expiration: change.isActive ? change.asActive.expiration.toString() : null,
             uploadedBy: tx.signer.inner.toHex(),
-          });
+            meta: null,
+          };
+
+          const metaHash = await getMetaHash(this.api.code, id.toHex());
+
+          if(metaHash) {
+            updateCodeInput.meta = await this.metaService.getByHashApi(metaHash);
+          }
+
+          codes.push(updateCodeInput);
         } else {
           const codeId = generateCodeHash(tx.args[0].toHex());
           const code = await this.codeRepository.get(codeId, this.genesis);
+          const metaHash = await getMetaHash(this.api.code, codeId);
```

Comment:
If you need to get meta under both conditions, I assume that it's better to do it once before if/else